src/PHAL_Dimension.hpp

```diff
@@ -85,4 +85,17 @@ namespace PHX {
   template<> struct is_extent<Dummy> : std::true_type {};
 }
 
+namespace PHX {
```

Comment:
Are these specialization declaration necessary? I thought ETI-ing the fcn in the cpp file was enough (much like we do for the evaluators)...

Reply:
I get this error if I don't  (Panzer does the same):
```
/ascldap/users/mperego/Workspace/albany/sources/albany-src/src/PHAL_Dimension.cpp:62:37: error: specialization of ‘std::string PHX::print() [with T = Dim; std::string = std::__cxx11::basic_string<char>]’ after instantiation
   62 |   template<> std::string print<Dim>(){return "Dim";}
```

Reply:
Ah, that's because you should put all the specializations at the top of the cpp file, before they are used in all the `name` methods below.

Btw, when I compile, all is good, but when I open the file in my editor, my background clang compiler issues a warning for all the `name()` implementations:
```
returning address of local temporary object
```

I think the reason is that `PHX::print<T>()` returns a string, which should get destroyed when `name()` returns, and yet, we return the pointer inside the std::string. I am not sure this is wrong, but I am also not sure it's right...

One way could be to do
```
static auto s = PHX::print<Node>();
return s.c_str();
``` 
It's a big cloggy, but definitely safe

Reply:
Ok, I confirmed this is indeed buggy with this:
```
std::string foo () {
  std::string a = "test me";
  return a;
}

const char* bar () {
  static auto s = foo();
  return s.c_str();
}

int main(int ,char** ){
  std::cout << "s: " << bar() << "\n";
  return 0;
}

```
Output with `static` in the impl of `bar`:

```
s: test me
```
Output without `static` in the impl of `bar`:
```
s:
```